pkg/artifactory/resource_artifactory_federated_repository_test.go

```diff
@@ -129,3 +132,109 @@ func TestAccFederatedRepoAllTypes(t *testing.T) {
 		})
 	}
 }
+
+func TestAccFederatedRepoWithProjectAttributesGH318(t *testing.T) {
+	if skip, reason := skipFederatedRepo(); skip {
+		t.Skipf(reason)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	projectKey := fmt.Sprintf("t%d", rand.Intn(100000000))
+	projectEnv := randProjectEnv()
+	repoName := fmt.Sprintf("%s-generic-federated", projectKey)
+
+	_, fqrn, name := mkNames(repoName, "artifactory_federated_generic_repository")
+	federatedMemberUrl := fmt.Sprintf("%s/artifactory/%s", os.Getenv("ARTIFACTORY_URL"), name)
+
+	params := map[string]interface{}{
+		"name":       name,
+		"projectKey": projectKey,
+		"projectEnv": projectEnv,
+		"memberUrl":  federatedMemberUrl,
+	}
+	federatedRepositoryConfig := executeTemplate("TestAccFederatedRepositoryConfig", `
+		resource "artifactory_federated_generic_repository" "{{ .name }}" {
+			key                  = "{{ .name }}"
+			project_key          = "{{ .projectKey }}"
+	 		project_environments = ["{{ .projectEnv }}"]
+
+			member {
+				url     = "{{ .memberUrl }}"
+				enabled = true
+			}
+		}
+	`, params)
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		PreCheck: func() {
+			testAccPreCheck(t)
+			createProject(t, projectKey)
+		},
+		CheckDestroy: verifyDeleted(fqrn, func(id string, request *resty.Request) (*resty.Response, error) {
+			deleteProject(t, projectKey)
+			return testCheckRepo(id, request)
+		}),
+		Steps: []resource.TestStep{
+			{
+				Config: federatedRepositoryConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "member.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "member.0.url", federatedMemberUrl),
+					resource.TestCheckResourceAttr(fqrn, "member.0.enabled", "true"),
+					resource.TestCheckResourceAttr(fqrn, "project_key", projectKey),
+					resource.TestCheckResourceAttr(fqrn, "project_environments.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "project_environments.0", projectEnv),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFederatedRepositoryWithInvalidProjectKeyGH318(t *testing.T) {
+	if skip, reason := skipFederatedRepo(); skip {
+		t.Skipf(reason)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	projectKey := fmt.Sprintf("t%d", rand.Intn(100000000))
```

Comment:
`randInt()` already does this exact thing

Reply:
No, it doesn't. Here I need to restrict the number to 9 digits only. `randInt()` can get above that.